stdlib/multiprocessing/resource_sharer.pyi

```diff
@@ -0,0 +1,36 @@
+import sys
+from collections.abc import Callable
+from socket import socket
+from typing import Union
+from typing_extensions import TypeAlias
+
+from .connection import Connection
+
+__all__ = ["stop"]
+
+# https://docs.python.org/3/library/multiprocessing.html#address-formats
+_Address: TypeAlias = Union[str, tuple[str, int]]
+
+if sys.platform == "win32":
+    __all__ += ["DupSocket"]
+
+    class DupSocket:
+        def __init__(self, sock: socket) -> None: ...
+        def detach(self) -> bytes: ...
+
+else:
+    __all__ += ["DupFd"]
+
+    class DupFd:
+        def __init__(self, fd: int) -> None: ...
+        def detach(self) -> None: ...
```

Comment:
It returns the first element whatever a call to `multiprocessing.reduction.recv_handle()` returns: https://github.com/python/cpython/blob/bceb197947bbaebb11e01195bdce4f240fdf9332/Lib/multiprocessing/reduction.py#L189

The current typeshed annotation for `multiprocessing.reduction.recv_handle()` (which was written by me :) is incorrect; I'll file a PR to fix it momentarily. I think it returns `list[int]`.

```suggestion
        def detach(self) -> int: ...
```